### PR TITLE
🚀 feat: initial commit of nightly build action

### PR DESCRIPTION
**Pull request by wh1337**
### What change does this PR introduce? 

RFC - Nightly Package Deployments
https://github.com/novuhq/novu/discussions/2062

### Why was this change needed?

Allows for users to get the latest changes from the `next` branch vs only the `main` branch.

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+          IMAGE_TAG: ${{ github.sha }}
+          GH_ACTOR: ${{ github.actor }}
+          GH_PASSWORD: ${{ secrets.GH_PACKAGES }}
+        run: |
+          echo $GH_PASSWORD | docker login ghcr.io -u $GH_ACTOR --password-stdin 
+          docker buildx build -t ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG -f apps/ws/Dockerfile .
+          docker tag ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:nightly
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:nightly
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG
+          echo "::set-output name=IMAGE::ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG"
```
Comment (LetItRock) [inline .github/workflows/nightly.yaml:308]: you've missed the widget

File: .github/workflows/nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  workflow_dispatch:
+
+  ## Set up a schedule for the workflow to run
+  schedule:
+    - cron: '0 2 * * *' # run at 0200 UTC
+
+jobs:
+  test_core:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (LetItRock) [inline .github/workflows/nightly.yaml:13]: question: shouldn't we check if there are any changes on the `next` branch before running this?

Comment (cryptic-ai) [reply .github/workflows/nightly.yaml:13]: It was my understanding to do it once a night versus on each merge with the next branch

We can change that if you guys want though!

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+      - name: Test
+        working-directory: 'packages/core'
+        run: pnpm test
+
+  test_api:
+    uses: ./.github/workflows/reusable-api-e2e.yaml
+
+  test_docs:
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (cryptic-ai) [reply .github/workflows/nightly.yaml:40]: Overlooked that, good catch. I'll get it updated tonight

File: .github/workflows/nightly.yaml (hunk 2 above)
Comment (cryptic-ai) [reply .github/workflows/nightly.yaml:13]: any update on if you want to do it on push to `next` or just nightly?

File: .github/workflows/nightly.yaml (hunk 1 above)
Comment (cryptic-ai) [reply .github/workflows/nightly.yaml:308]: With the holidays being done, I will get to work on this this week.

**Comment by scopsy** [comment]
@wh1337 thank you so much for the effort on this, but at the current stage, the diff is quite big. And we would need to rething this a bit. I'm closing this PR for now due to inactivity and priority shift, and if we will get back to it, will add you as a collaborator on the new implementation so we can use the ideas and code parts from this PR. 

Hope it didn't create too much inconvinience for you as you invested the time in this. 🙏♥️

**Pull request closed without merging**